projects/trenchboot-aem.md: add application with giscus system (#122)
Dasharo/docs: docs/projects/trenchboot-aem.md

for operating system attestation. However, DRT technologies are designed in a
way to provide the ability to establish a secure environment for integrity
measurements at any arbitrary point of time instead of relying on the firmware
which requires machine reset to establish the aforementioned secure
environment.

The usage of DRT technologies like Intel Trusted Execution Technology (TXT) or
AMD Secure Startup becomes more and more significant, for example, Dynamic Root
of Trust for Measurement (DRTM) requirements of
[Microsoft Secured Core PCs][ms_sec_core]. In open-source projects, DRTM hasn't
found its place yet, but that gradually changes. The demand on having firmware
independent Roots of Trust is increasing and projects that satisfy this demand
are growing, for instance, [TrenchBoot][tb_org]. TrenchBoot is a framework that
allows individuals and projects to build security engines to perform launch
integrity actions for their systems. The framework builds upon Boot Integrity
Technologies (BITs) that establish one or more Roots of Trust (RoT) from which
a degree of confidence that integrity actions were not subverted. The project

> has grown a lot thanks to the previous [NLnet NGI0 PET grant][nlnet_opendrtm]

Reviewer: ```suggestion
has grown a lot thanks to the previous [NLnet NGI0 PET grant][nlnet_open_drtm]
```

   technology. Implementing AMD support would make a huge impact and broaden
   the usage of DRTM technologies.

The initial AEM implementation relied on the [Trusted Boot][tboot_wiki],
Intel's reference implementation of Intel TXT. It had never any plans to
support AMD processors. TrenchBoot is filling this gap supporting both Intel
and AMD hardware which makes it an ideal target to replace Trusted Boot in
Qubes OS AEM implementation. Furthermore, the project grant would be used to
implement the missing pieces in the Qubes OS AEM software to cover the AMD and
Intel support for both TPM 1.2 and TPM 2.0.

## Compare your own project with existing or historical efforts

3mdeb is a licensed provider for quality coreboot consulting services for 5
years. We are well-known in the open-source community for maintaining firmware
of the PC Engines APU series platform for over 4-years. Delivering high-quality
firmware releases each month and providing technical support on PC Engines and
OPNSense forums. 3mdeb embedded systems developers are experienced engineers
accustomed to operating systems development. Our developers have contributed to

> the [fwupd support for Qubes OS][8]. 3mdeb is also regularly co-organizing

Reviewer: ```suggestion
the [fwupd support for Qubes OS][qubes_fwupd]. 3mdeb is also regularly co-organizing
```

to date does not prevent the firmware vulnerabilities to appear. The Static
Root of Trust (SRT) like Unified Extensible Firmware Interface (UEFI) Secure
Boot and measured boot provided by the firmware is not always sufficient to
establish a secure environment for an operating system. In case the firmware is
compromised, it could inject malicious software into operating system
components and prevent the machine owner from detecting it. Silicon vendors
implement alternative technologies to establish a Dynamic Root of Trust (DRT)
to provide a secure environment for operating system launch and integrity
measurements. These integrity measurements either from SRT or DRT can be used
for operating system attestation. However, DRT technologies are designed in a
way to provide the ability to establish a secure environment for integrity
measurements at any arbitrary point of time instead of relying on the firmware
which requires machine reset to establish the aforementioned secure
environment.

The usage of DRT technologies like Intel Trusted Execution Technology (TXT) or
AMD Secure Startup becomes more and more significant, for example, Dynamic Root

> of Trust for Measurement (DRTM) requirements of Microsoft Secured Core PCs

Reviewer 2: I think we should [labels](https://spec.commonmark.org/0.30//#link-reference-definitions) or just link, why we keep things in so weird way? Current format cannot be transferred easily to MoU.

```suggestion
of Trust for Measurement (DRTM) requirements of [Microsoft Secured Core PCs](https://some.link)
```

Author: Done for all links using labels

   effectively limits the usage of AEM on most if not all modern machines.
3. Qubes OS AEM has never supported any AMD processors with AMD Secure Startup
   technology. Implementing AMD support would make a huge impact and broaden
   the usage of DRTM technologies.

The initial AEM implementation relied on the Trusted Boot project \[[7]\],
Intel's reference implementation of Intel TXT. It had never any plans to
support AMD processors. TrenchBoot is filling this gap supporting both Intel
and AMD hardware which makes it an ideal target to replace Trusted Boot in
Qubes OS AEM implementation. Furthermore, the project grant would be used to
implement the missing pieces in the Qubes OS AEM software to cover the AMD and
Intel support for both TPM 1.2 and TPM 2.0.

\[1\]: `https://docs.microsoft.com/en-us/windows-hardware/design/device-experiences/oem-highly-secure#what-makes-a-secured-core-pc`

\[2\]: `https://trenchboot.org/`

\[3\]: `https://nlnet.nl/project/OpenDRTM/`

\[4\]: `https://www.qubes-os.org/`

\[5\]: `https://blog.invisiblethings.org/2011/09/07/anti-evil-maid.html`

\[6\]: `http://theinvisiblethings.blogspot.com/2009/10/evil-maid-goes-after-truecrypt.html`

> \[7\]: `https://sourceforge.net/p/tboot/wiki/Home/`

Reviewer 2: This is redundant

Author: Removed

         + allocate the memory space for the trampoline used to launch other
           processors or use the allocation done by GRUB2 if necessary
         + do not go error path when Boot Services are not present and skip all
           calls to UEFI Boot Services by using the information provided by GRUB2

    - Test the solution on AMD and Intel hardware with TPM 2.0 and TPM 1.2 with
      legacy and UEFI boot mode

## Projects or organizations relevant to this project before?

1. [Qubes OS](https://www.qubes-os.org/)
2. [Xen Hypervisor](https://xenproject.org/)
3. [GNU GRUB](https://www.gnu.org/software/grub/)
4. [TrenchBoot](https://trenchboot.org/)
5. [Invisible Things Lab](https://invisiblethingslab.com/)
6. [Apertus Solutions](https://apertussolutions.com/)
7. [Oracle](https://www.oracle.com/)
8. [3mdeb](https://3mdeb.com/)

## The ecosystem of the project

3mdeb has a good relationship with the maintainers of relevant projects which
will participate in review of the work:

- Marek Marczykowski-Górecki (Invisible Things Lab CTO) - Qubes OS maintainer
- Andrew Cooper (Citrix) - Xen Hypervisor Maintainer
- Daniel Kiper (Oracle) - GRUB2 Maintainer

> - Daniel Smith (Apertus Solutions) - TrenchBoot founder and maintainer

Reviewer 2: Why not say how further review will be continued? That people can use Giscus or that they can contribute directly to this repository.

Author: Done
